tools/lsp/properties.rs

```diff
@@ -685,32 +690,30 @@ fn find_insert_position_for_property(
         && (property_index - previous_index <= next_index - property_index)
     {
         // There is a previous index and it is closer than the next one
-        Some(
-            properties
-                .get(previous_index)
-                .expect("Index was checked to exist")
-                .defined_at
-                .as_ref()
-                .expect("This index had a defined_at")
-                .delete_range
-                .end
-                .clone(),
-        )
+        let p = properties
+            .get(previous_index)
+            .expect("Index was checked to exist")
+            .defined_at
+            .as_ref()
+            .expect("This index had a defined_at")
+            .delete_range
+            .end
+            .clone();
```

Comment:
this does a lot of clone  (here and on next line?)

Also, this is error prone, many expect.  Why not keep the ranges in Option, instead of the previous_index / next_index.


But tbh, i think this should be done by looking at the syntax_nodes::Element for the given element, instead of looking at the `PropertyInformation`
You can use  `syntax_nodes::Element::Binging()` and `syntax_nodes::Element::PropertyDeclaration()` to find the existing ones, and add yours after the last of these, or after the SyntaxKind::LBrace.